### PR TITLE
Added overlay selectors

### PR DESCRIPTION
**Pull request by Blackbaud-AlexKingman**

### CODE REVIEW

File: src/app/public/test-utility/test-utility.spec.ts [hunk 1]
```diff
@@ -38,6 +45,63 @@ class TestParentComponent { }
...
+@NgModule({
+  declarations: [
+    TestOverlayComponent,
+    TestOverlayInternalComponent
+  ],
+  entryComponents: [
+    TestOverlayInternalComponent
+  ]
+})
+class OverlayTestModule {}
```
Comment (Blackbaud-AlexKingman) [inline src/app/public/test-utility/test-utility.spec.ts:103]: ☝️  I considered breaking these out into their own file... There were already a couple test components in here so I left them in. I could be persuaded if the reviewer thinks its worth it.

File: src/app/public/test-utility/test-utility.ts [hunk 2]
```diff
@@ -146,4 +146,49 @@ export class SkyAppTestUtility {
     return skyEl;
   }
 
+  /**
+   * Returns the first element inside the overlay that is a match of the
+   * provided selector. Selector should not contain SKYUX prefixes, as they are for
+   * internal use only and are subject to change.
+   */
+  public static overlayQuerySelector(selector: string): any {
+    if (selector.substring(0, 5) === '.sky-' || selector.substring(0, 4) === 'sky-') {
```
Comment (Blackbaud-SteveBrush) [inline src/app/public/test-utility/test-utility.ts:155]: What about selectors that don't start with `.sky-`? Such as `#foobar .sky-colorpicker`? You could just do an `indexOf` check...

File: src/app/public/test-utility/test-utility.ts [hunk 3]
```diff
@@ -146,4 +146,49 @@ export class SkyAppTestUtility {
...
+      throw new Error(`Selector should not contain SKYUX prefixes.`);
```
Comment (Blackbaud-SteveBrush) [inline src/app/public/test-utility/test-utility.ts:156]: "SKY UX"

File: src/app/public/test-utility/test-utility.ts (hunk 2 above)
Comment (Blackbaud-AlexKingman) [reply src/app/public/test-utility/test-utility.ts:155]: Good call.

Comment (Blackbaud-AlexKingman) [reply src/app/public/test-utility/test-utility.ts]: Fixed.

File: src/app/public/test-utility/test-utility.ts (hunk 3 above)
Comment (Blackbaud-AlexKingman) [reply src/app/public/test-utility/test-utility.ts]: Fixed.

**Comment by Blackbaud-AlexKingman** [comment]
After some further discussion, we decided adding `querySelector` to the public API would add confusion of when to use it vs the Web API's `querySelector`. This logic will be handled in each component fixture. Closing.

**Pull request closed without merging**